### PR TITLE
Extract function components from sidebar

### PR DESCRIPTION
**Pull request by eldano**
This is just a proposal so we can discuss it.
Pros:
- separation of concerns

Cons:
- html and css structure is spread in multiple functions (maybe it's a consequence of using tailwind?)
- it's hard to follow the data in the assigns (maybe we already had this problem? see `@contextual_help`
- now we have some template code in heex files and some in ~H sigils (maybe we can move everything to sigils if template code is not too large?)

### CODE REVIEW

**Comment by jmbejar** [comment]
> Take this as a question, not a strong opinion at all... Would it feel more natural if everything is moved into the elixir module? I mean, having most of the markup code in inline functions but also keeping the separate hex file looks like some back and forth indirection between files to me... just a feeling.
> 
> Implementing what is now left in the separate file in the `render` function is the way to complete this refactor?

Maybe this is an example of this https://github.com/livebook-dev/livebook/blob/main/lib/livebook_web/live/home_live.ex
(see `sessions_list` private function called from render)

**Comment by eldano** [comment]
> Take this as a question, not a strong opinion at all... Would it feel more natural if everything is moved into the elixir module? I mean, having most of the markup code in inline functions but also keeping the separate hex file looks like some back and forth indirection between files to me... just a feeling.

Yes I agree. I don't like that much the example you gave because the render function is huge, but separating parts in different functions and having the `render` function as the base sounds like the best approach. I'll complete this PR with that change so we can see how it looks.

Also I see we could pattern match on the `bindings` function to remove the `if` in the template.
